components/mpas-ocean/src/shared/mpas_ocn_tracer_hmix_redi.F

```diff
@@ -240,6 +247,9 @@ subroutine ocn_tracer_hmix_Redi_tend(meshPool, layerThickness, layerThicknessEdg
                redi_term1(iTr, k, iCell) = redi_term1(iTr, k, iCell) - (1.0_RKIND + term1TaperFactor* &
                                            (kappaRediEdgeVal - 1.0_RKIND))*edgeSignOnCell(i, iCell)*flux*invAreaCell
 
+               if (.not.use_Redi_diag_terms) cycle
```

Comment:
@philipwjones any performance comments on my method here would be appreciated. I use a local logical variable and cycle just before computations involving terms 2 and 3. That means I have an if statement above, but that seems better than keeping an additional nCell-sized mask in memory.

Reply:
These loops are complex enough that you're probably not getting much optimization anyway, so this doesn't do much harm.  I was actually about to re-do this routine for GPU optimization before I got distracted by the vmix routines, so I'll see what I can do with it later.